File: python/sparklabs/batch/sales/dataset/spark_salting.py
from sparklabs.batch.sales.dataset import utils


def run() -> None:
    (
        utils.spark.range(20)
        .rdd.map(lambda row: (row["id"], int(__import__("random").random() * 10)))
        .toDF(["id", "salt"])
        .show(truncate=False)
    )


if __name__ == "__main__":
    run()

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/dataset/SparkSalting.scala
package io.devorbit.sparklabs.batch.sales.dataset

import io.devorbit.sparklabs.batch.sales.dataframe.Utils

object SparkSalting extends App {
  import Utils.sparkSession.implicits._

  Utils.sparkSession.range(20).as[Long]
    .map(id => (id, (scala.util.Random.nextDouble() * 10).toInt))
    .toDF("id", "salt")
    .show(false)
}
